File: 0485 - Max Consecutive Ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        consecutives = [0] * (len(nums) + 1)
        max_consecutive = 0
        for index, num in enumerate(nums):
            if num:
                consecutives[index + 1] = consecutives[index] + 1
                max_consecutive = max(max_consecutive, consecutives[index + 1])
            else:
                consecutives[index] = 0
        return max_consecutive

File: 0485 - Max Consecutive Ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int max_consecutive = nums[0];
        for(int i = 1; i < nums.size(); i++){
            if(nums[i] == 1){
                nums[i] = nums[i - 1] + 1;
                max_consecutive = max(nums[i], max_consecutive);
            }
        }
        return max_consecutive;
    }
};